### PR TITLE
chore(litmus-portal): Mapping workflow run to workflow schedules using workflow schedule id.

### LINKED ISSUE #1977: Map workflow runs to workflow schedules for litmus portal.
Currently, there is no mapping / relationship of a workflow schedule to a workflow run in litmus-portal, A mapping function is needed to embed generated workflow id into argo workflow manifest for indexing workflow runs based on their schedules.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
Signed-off-by: ishangupta-ds <ishan.gupta@mayadata.io>

## Proposed changes

Currently, there is no mapping of a workflow schedule to a workflow run, this PR embeds workflow id into workflow manifest before mutation for the subscriber to extract it and send to the gql server again for mapping after workflow run mutation.

fixes #1977

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
High priority PR to be merged before database normalization.

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@rajdas98 PTAL.

File: litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go [hunk 1]
```diff
@@ -151,9 +150,15 @@ func CreateChaosWorkflow(input *model.ChaosWorkFlowInput, r store.StateData) (*m
...
+
+	updatedWorkflowManifest, err := utils.EmbedWorkflowIDIntoManifest(workflow_id, input.WorkflowManifest)
+	if err != nil {
+		return nil, err
+	}
+
 	newChaosWorkflow := database.ChaosWorkFlowInput{
 		WorkflowID:          workflow_id,
-		WorkflowManifest:    input.WorkflowManifest,
+		WorkflowManifest:    updatedWorkflowManifest,
```
Comment (rajdas98) [inline litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go:161]: Use this pkg `github.com/tidwall/sjson`. You modified it in a single line.
```
	newWorkflowManifest, _ := sjson.Set(input.WorkflowManifest, "metadata.labels.0.workflow_id", workflow_id)
```

File: litmus-portal/backend/auth/go.sum [hunk 2]
```diff
@@ -13,19 +13,25 @@ github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
 github.com/dgrijalva/jwt-go v3.2.0+incompatible h1:7qlOGliEKZXTDg6OTjfoBKDXWrumCAMpl/TFQ4/5kLM=
 github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
```
Comment (rajdas98) [inline litmus-portal/backend/auth/go.sum:14]: Can you do `go mod tidy` and push the changes?

File: litmus-portal/frontend/src/utils/yamlUtils.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ const parsed = (yaml: string) => {
     const parsedYaml = YAML.parse(file as string);
     try {
       const embeddedYaml =
-        parsedYaml.spec.workflowSpec.templates[1].inputs.artifacts[0].raw.data;
```
Comment (rajdas98) [inline litmus-portal/frontend/src/utils/yamlUtils.ts:32]: Why thus change required?

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 4]
```diff
@@ -1,31 +1,16 @@
 export default [
```
Comment (rajdas98) [inline litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:1]: Please raise this changes in a different PR.

**Review by rajdas98 — APPROVED** [review]

File: litmus-portal/frontend/src/utils/yamlUtils.ts (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/yamlUtils.ts:32]: The parsing is only possible for workflows not cron workflows so we have to change it to test this.

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 4 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:1]: It is required..

**Comment by rajdas98** [comment]
Also, change wfid to workflow_id in subscriber

File: litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go]: Have updated it, with a minor change, using `metadata.labels.workflow_id`

File: litmus-portal/frontend/src/utils/yamlUtils.ts (hunk 3 above)
Comment (rajdas98) [reply litmus-portal/frontend/src/utils/yamlUtils.ts:32]: Currently, we don't have cron workflow Right? You can raise it in a separate PR.

**Comment by ishangupta-ds** [comment]
@rajdas98 PTAL.

**Comment by ishangupta-ds** [comment]
@rajdas98 PTAL.

File: litmus-portal/frontend/src/utils/yamlUtils.ts (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/yamlUtils.ts:32]: Yeah this is required as the parser was for cron workflow, i have changed it to normal workflow, without this change the frontend will break. @rajdas98

**Comment by ishangupta-ds** [comment]
> Also, change wfid to workflow_id in subscriber

Done.

File: litmus-portal/backend/auth/go.mod [hunk 5]
```diff
@@ -3,7 +3,6 @@ module github.com/litmuschaos/litmus/litmus-portal/backend/auth
 go 1.13
 
 require (
-	github.com/99designs/gqlgen v0.11.3
```
Comment (rajdas98) [inline litmus-portal/backend/auth/go.mod:6]: Why are you removing "	github.com/99designs/gqlgen v0.11.3" and "go.mongodb.org/mongo-driver"

Comment (ishangupta-ds) [reply litmus-portal/backend/auth/go.mod:6]: Just did a go mod tidy as you asked..it automatically got removed.

**Review by rajdas98 — APPROVED** [review]

**Comment by ishangupta-ds** [comment]
[](url)
Screenshot 2020-09-06 at 7 34 41 PM

File: litmus-portal/backend/auth/go.mod (hunk 5 above)
Comment (rajdas98) [reply litmus-portal/backend/auth/go.mod:6]: Okay, these changes are in auth

**Pull request merged**